src/hotspot/share/cds/heapShared.hpp

```diff
@@ -138,21 +139,90 @@ class ArchivedKlassSubGraphInfoRecord {
 };
 #endif // INCLUDE_CDS_JAVA_HEAP
 
+struct LoadedArchiveHeapRegion;
+
 class HeapShared: AllStatic {
   friend class VerifySharedOopClosure;
- private:
 
+public:
+  // At runtime, heap regions in the CDS archive can be used in two different ways,
+  // depending on the GC type:
+  // - Mapped: (G1 only) the regions are directly mapped into the heap
+  // - Loaded: At VM start-up, the objects in the heap regions are copied into the
+  //           heap's old generation. This is easier to implement than mapping but
```

Comment:
I'd refrain from saying "heap's old generation", as the interface apparently gives the GC freedom in putting the things where it can.

Reply:
I've updated the comments.